http-api/src/lib.rs

```diff
@@ -76,6 +79,44 @@ impl Context {
 
         Ok(())
     }
+
+    /// Populate a map between ssh fingerprints and their peer identities
+    fn populate_fingerprints(
+        &self,
+        map: &mut HashMap<String, Peer>,
+        urn: Urn,
+    ) -> Result<(), Error> {
+        use librad::git::identities::SomeIdentity;
+
+        let storage = ReadOnly::open(&self.paths).map_err(Error::from)?;
+        let project = identities::project::get(&storage, &urn)
+            .map_err(Error::Identities)?
+            .ok_or(Error::NotFound)?;
+        let meta: project::Metadata = project.try_into()?;
+
+        let identities = identities::any::list(&storage)?;
+        for identity in identities.flatten() {
+            if let SomeIdentity::Person(person) = identity {
+                if let Some(pk) = person.delegations().iter().next() {
```

Comment:
Convert this into a `while let` loop because we want *all* delegations of a person here, ie. if they have multiple keys we want all their keys to point to their identity.

Reply:
hmm it seems to be able to do a `while let` over an iterator, one has to return a Result variant..
Another option is to iterate with `for pk in person.delegations().iter()` wdyt? It seems to me that it should do the same and is maybe also more descriptive.

Reply:
`while let` works with any iterator, but you have to create the iterator separately and then call `next()`. Using a `for in` loop is perfectly fine as well.